### ISSUE TITLE
Searching by index not work for doc for Qt Assistant

### ISSUE THREAD

**Issue by playgithub** [doxygen/doxygen#9253]
**Describe the bug**
qch (qt help) searching by index doesn't work

**Expected vs observed behaviour**
Searching by index in Qt Assistant should work properly

**To Reproduce**
1. Build qch<br>
```sh
$ git clone --depth 1 https://github.com/wxWidgets/wxWidgets.git
$ cd wxWidgets/docs/doxygen/
$ WX_SKIP_DOXYGEN_VERSION_CHECK=1 ./regen.sh qch
warning: Tag 'LATEX_SOURCE_CODE' at line 459 of file 'Doxyfile' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'DOCBOOK_PROGRAMLISTING' at line 501 of file 'Doxyfile' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'CLASS_DIAGRAMS' at line 552 of file 'Doxyfile' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
Building up file structure...
Insert custom filters...
Insert help data for filter section (1 of 1)...
Insert files...
Warning: Cannot open file /path/to/wxWidgets/docs/doxygen/out/html/generic, skipping it...
Insert contents...
Insert indices...
Documentation successfully generated.
```
2. Add `wxWidgets/docs/doxygen/out/wx.qch` to `qt assistant` and search s.th. like `wxButton` by index, the result is not as expected as below
![image](https://user-images.githubusercontent.com/3665590/161730983-f8c2b52e-ffa0-4207-b608-001490c52784.png)

**More Log**
doxygen.log
```
docs/msw/gtk.md:23: warning: Found unknown command '\bakefiles'
docs/msw/gtk.md:23: warning: Found unknown command '\bakefiles'
docs/doxygen/overviews/resyntax.h:651: warning: found </center> tag without matching <center>
docs/doxygen/overviews/xrc_format.h:2953: warning: Found unknown command '\t'
docs/doxygen/overviews/xrc_format.h:2953: warning: Found unknown command '\r'
docs/doxygen/overviews/xrc_format.h:2971: warning: end of comment block while expecting command </span>
docs/doxygen/overviews/resyntax.h:651: warning: found </center> tag without matching <center>
docs/doxygen/overviews/xrc_format.h:2953: warning: Found unknown command '\t'
docs/doxygen/overviews/xrc_format.h:2953: warning: Found unknown command '\r'
docs/doxygen/overviews/xrc_format.h:2971: warning: end of comment block while expecting command </span>
docs/msw/gtk.md:23: warning: Found unknown command '\bakefiles'
docs/msw/gtk.md:23: warning: Found unknown command '\bakefiles'
interface/wx/propgrid/property.h:580: warning: Found unknown command '\r'
interface/wx/webview.h:809: warning: Unsupported xml/html tag <name> found
```

**Platform and version information**
 - doxygen: 1.9.3
 - Qt Assistant: 5.15.3
 - wxWidgets version you use: master/HEAD, commit id: 4da102f2b1
 - OS and its version: Manjaro

**More Info**
Last year the qch file generated with the same warnings worked fine.

**Comment by albert-github** [comment]
You wrote:
> Last year the qch file generated with the same warnings worked fine.

- Which version was this?
   -  did you also try the current master (i.e. 1.9.4 (5b7315a52bb1229875cad513e9da1263e6d934a7), my impression is that it is better)?
- what is the exact output of the `doxygen -v`?
- The example is a bit big and is a moving target (wxWidget repository can change): Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
- which version of the Qt Assistent and qhelpgenerator are you using?

Note the warnings reported in the `doxygen.log` shouldn't influence the mentioned Qt help problem you have.

**Comment by playgithub** [comment]
> You wrote:
> 
> > Last year the qch file generated with the same warnings worked fine.
> * Which version was this?

I can't remember, just a hint.

>   * did you also try the current master (i.e. 1.9.4 ([5b7315a](https://github.com/doxygen/doxygen/commit/5b7315a52bb1229875cad513e9da1263e6d934a7)), my impression is that it is better)?

Not yet, because packages in the official repo is much easy to maintain, can wait for 1.9.4

> * what is the exact output of the `doxygen -v`?
```sh
$ doxygen -v
1.9.3
```

> * The example is a bit big and is a moving target (wxWidget repository can change): Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

It's efficient for git to clone just one depth.
```sh
$ git clone --depth 1 https://github.com/wxWidgets/wxWidgets.git
```

> * which version of the Qt Assistent and qhelpgenerator are you using?
no `qthelpgenerator`
```
$ whereis qthelpgenerator
qthelpgenerator:
```
In my experience, `qthelpgenerator` is not necessary.

**Comment by playgithub** [comment]
> * `git clone` is still a moving target

Here is a static one
[wxWidgets.zip](https://github.com/doxygen/doxygen/files/8418438/wxWidgets.zip)

> * "In my experience, qthelpgenerator is not necessary.": How do you go from qhp file to qch file?

I'm not sure if it has been installed, however `qch` has been generated, maybe it has another name in `Manjaro` repo.

**Comment by albert-github** [comment]
It looks like this problem is solved as a side effect of some other fixes, to get it properly noted for the next release I'm reopening the issue (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
